File: app/src/main/java/com/stathis/elmepaunivapp/ui/dashboard/recyclerview/DashboardAdapter.java
package com.stathis.elmepaunivapp.ui.dashboard.recyclerview;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;
import androidx.recyclerview.widget.ListAdapter;

import com.stathis.elmepaunivapp.R;
import com.stathis.elmepaunivapp.abstraction.DiffItemCallbackClass;
import com.stathis.elmepaunivapp.listeners.DashboardOptionListener;
import com.stathis.elmepaunivapp.listeners.DepMembersClickListener;
import com.stathis.elmepaunivapp.ui.dashboard.model.DashboardOption;
import com.stathis.elmepaunivapp.ui.department.recyclerviews.DepMembersViewHolder;

public class DashboardAdapter extends ListAdapter<DashboardOption,DashboardViewHolder> {

    private DashboardOptionListener listener;

    public DashboardAdapter(DashboardOptionListener listener) {
        super(new DiffItemCallbackClass<DashboardOption>());
        this.listener = listener;
    }

    @NonNull
    @Override
    public DashboardViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.holder_dashboard_option, parent, false);
        return new DashboardViewHolder(view, listener);
    }

    @Override
    public void onBindViewHolder(@NonNull DashboardViewHolder holder, int position) {
        holder.present(getItem(position));
    }
}

File: app/src/main/java/com/stathis/elmepaunivapp/ui/home/fragments/dashboard/recyclerview/DashboardAdapter.kt
package com.stathis.elmepaunivapp.ui.home.fragments.dashboard.recyclerview

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import com.stathis.elmepaunivapp.abstraction.DiffItemClass
import com.stathis.elmepaunivapp.callbacks.ElmepaClickListener
import com.stathis.elmepaunivapp.abstraction.LocalModel
import com.stathis.elmepaunivapp.databinding.HolderDashboardOptionBinding

class DashboardAdapter(private val callback: ElmepaClickListener) :
    ListAdapter<LocalModel, DashboardViewHolder>(DiffItemClass<LocalModel>()) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DashboardViewHolder {
        val view = HolderDashboardOptionBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return DashboardViewHolder(view, callback)
    }

    override fun onBindViewHolder(holder: DashboardViewHolder, position: Int) {
        holder.bindData(getItem(position))
    }
}